src/payment/strategies/adyenv2/adyenv2-payment-strategy.ts

```diff
@@ -31,33 +32,38 @@ export default class AdyenV2PaymentStrategy implements PaymentStrategy {
         private _locale: string
     ) {}
 
-    initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+    async initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
         const { adyenv2 } = options;
 
         if (!adyenv2) {
             throw new InvalidArgumentError('Unable to initialize payment because "options.adyenv2" argument is not provided.');
         }
 
-        const paymentMethod = this._store.getState().paymentMethods.getPaymentMethod(options.methodId);
+        this._paymentInitializeOptions = adyenv2;
 
-        if (!paymentMethod) {
-            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
-        }
+        const paymentMethod = this._store.getState().paymentMethods.getPaymentMethodOrThrow(options.methodId);
 
-        this._adyenv2 = adyenv2;
-
-        const configuration: AdyenConfiguration = {
+        this._adyenClient = await this._scriptLoader.load({
             environment:  paymentMethod.initializationData.environment,
             locale: this._locale,
             originKey: paymentMethod.initializationData.originKey,
             paymentMethodsResponse: paymentMethod.initializationData.paymentMethodsResponse,
-        };
+        });
 
-        this._scriptLoader.load(configuration)
-            .then(adyenCheckout => {
-                this._adyenCheckout = adyenCheckout;
-                this._mountComponent(paymentMethod.method);
-            });
+        try {
+            this._paymentComponent = await this._mountPaymentComponent(paymentMethod);
+        } catch (error) {
+            throw error;
```

Comment:
If you're not handling the error here, you don't need a `try/catch`. You can just let the error bubble up.